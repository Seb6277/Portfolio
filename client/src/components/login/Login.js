import React from 'react'
import {Button, FormGroup, Form, Label} from 'reactstrap'

class Login extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      email: "",
      password: ""
    }
  }

  validateForm() {
    return this.state.email.length > 0 && this.state.password.length > 0
  }

  handleMailChange = event => {
    this.setState({email: event.target.value})
  }

  handlePasswordChange = event => {
    this.setState({password: event.target.value})
  }

  handleSubmit(event) {
    event.preventDefault()
    this.isValidCredential().then(() => {
      localStorage.setItem("authToken", "authToken")
      document.location.href="/admin"
    }).catch(error => {
      console.error(error)
    })
  }

  isValidCredential() {
    return new Promise((resolve, reject) => {
      if (this.state.email === process.env.REACT_APP_MAIL && this.state.password === process.env.REACT_APP_PASSWD) {
        resolve(true)
      } else {
        reject("Bad Credential")
      }
    })
  }

  render() {
    return(
      <div className="Login col-md-8 offset-md-2">
        <h1>Login</h1>
        <Form>
          <FormGroup>
            <Label for="email">Email :</Label><br/>
            <input type="email" value={this.state.email} onChange={this.handleMailChange}/>
          </FormGroup>
          <FormGroup>
            <Label for="password">Password :</Label><br/>
            <input type="password" value={this.state.password} onChange={this.handlePasswordChange}/>
          </FormGroup>
          <Button
            type="submit"
            disabled={!this.validateForm()}
            onClick={(e) => this.handleSubmit(e)}
          >
            Login
        </Button>
        </Form>
      </div>
    )
  }
}

export default Login