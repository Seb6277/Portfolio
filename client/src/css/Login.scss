.Login{
  text-align: center;
  margin-top: 30px;
  input{
    width: 70%;
  }
}

@media screen and (max-width: 640px){
  .Login{
    margin-right: 10px;
    margin-left: 10px;
  }
}