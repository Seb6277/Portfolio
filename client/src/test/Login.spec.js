import Login from '../components/login/Login'

describe('<Login/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Login/>)
    expect(wrapper).toMatchSnapshot()
  });
})