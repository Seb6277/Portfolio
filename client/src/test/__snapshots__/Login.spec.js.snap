// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render correctly 1`] = `
<div
  className="Login col-md-8 offset-md-2"
>
  <h1>
    Login
  </h1>
  <Form
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        for="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email :
      </Label>
      <br />
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        for="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password :
      </Label>
      <br />
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormGroup>
    <Button
      color="secondary"
      disabled={true}
      onClick={[Function]}
      tag="button"
      type="submit"
    >
      Login
    </Button>
  </Form>
</div>
`;
